add https readme (#10315)
istio/istio: samples/https/README.md

# HTTPS service

> This sample consists of an NGINX service which can encrypt traffic using HTTPS.

Reviewer: I would remove which can encrypt traffic and leave just: _This sample defines an HTTPS service backed by NGINX._

Author: fixed. Thanks.

Reviewer 2: Doesn't seem fixed?

To use it:

1. Generate certificates

>     You need to have openssl installed to run these commands:

Reviewer 3: ```suggestion
    You need to have [OpenSSL](https://www.openssl.org/) installed to run these commands:
```

Reviewer 4: You need to install [OpenSSL](https://www.openssl.org/) if not already installed:

>     `openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"`

Reviewer 3: ```suggestion
    ```console
    $ openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /tmp/nginx.key -out /tmp/nginx.crt -subj "/CN=my-nginx/O=my-nginx"
    ```
```

>     `kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt`

Reviewer 3: ```suggestion
    ```console
    $ kubectl create secret tls nginxsecret --key /tmp/nginx.key --cert /tmp/nginx.crt
    ```
```

> 2. Create a configmap used for the HTTPS service

Reviewer 3: ```suggestion
2. Create a `ConfigMap` used for the HTTPS service
```

>     `kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf`

Reviewer 3: ```suggestion
    ```console
    $ kubectl create configmap nginxconfigmap --from-file=samples/https/default.conf
    ```
```

3. Deploy a NGINX-based HTTPS service

>     `kubectl apply -f samples/https/nginx-app.yaml`

Reviewer 3: ```suggestion
    ```console
    $ kubectl apply -f samples/https/nginx-app.yaml
    ```
```

> Call my-nginx service `curl https://my-nginx -k` to verify this creation.

Reviewer 3: ```suggestion
Send a request to the `my-nginx` service over HTTPS using the following command.

    ```console
    $ curl https://my-nginx -k
    ```
```
